OpenRA.Mods.Common/Effects/RevealShroudEffect.cs

```diff
@@ -0,0 +1,90 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2016 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Effects;
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Effects
+{
+	public class RevealShroudEffect : IEffect
+	{
+		static readonly PPos[] NoCells = { };
+
+		WPos pos;
+		Player player;
+		Shroud.SourceType sourceType;
+		int remainingTicks;
```

Comment:
You're using three variables here (`remainingDelay`, `cellsRevealed`, `remainingTicks`) when you only really need one incrementing `tick` that starts with a value of `-delay`,  reveals the shroud at `0`, and removes the shroud at `duration`.